### PR TITLE
🚧📰 Groovy => Kotlin conversion + Version Catalog

### LINKED ISSUE #43: Update dependencies (Compose, Kotlin and more)
## Description
There are a bunch of new library versions we should update, most importantly Compose and Kotlin.

## Definition of done
- [ ] Kotlin is updated
- [ ] Compose is updated and included using bom
- [ ] Any other low hanging fruit

### LINKED ISSUE #44: Convert Groovy gradle files to Kotlin
## Description
Let's get rid of groovy

## Definition of done
- [ ] All groovy files are now .kts

### LINKED ISSUE #45: Use gradle dependency catalog
## Description
We should be using dependency catalog to share dependencies https://docs.gradle.org/current/userguide/platforms.html

## Definition of done
- [ ] Add `libs.versions.toml` file and use it

### LINKED ISSUE #47: Disable Jetifier
## Description
It is time
`android.enableJetifier=false`

### PR DESCRIPTION
**Pull request by semanticer**
## Summary
Remember when they told you that you should split the work into smaller chunks? Well, this PR isn't the best example of this approach 😅 

It just felt easier to do the groovy to kotlin transition together with the gradle version catalog thingy. 

I just wanted to point out a few things we gained from this transition: (especially in Giraffe which supports version catalogs natively)

Autocompletion for dependencies:

Screenshot 2023-04-11 at 16 08 27

Better dependency checks:

Screenshot 2023-04-11 at 15 54 48

Versions checks in AS for free:

Screenshot 2023-04-11 at 15 54 05

Cool right? 

I'm also making `0.2.1` release from this branch to prove that releasing still works as it should.

## Relationships

Closes #47
Closes #45 
Closes #44
Closes #43

## Review pointers
We need to make sure that we can still:
- build the app
- run tests successfully 
- run spotless 
- publish the library

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        classpath(libs.junit5.gradlePlugin)
+    }
+}
+
+plugins {
+    alias(libs.plugins.spotless)
+    alias(libs.plugins.nexus)
+}
+
+extra.apply {
```
[the remark this answers is not in the document]
Comment (semanticer) [reply build.gradle.kts:15]: Me too :-D

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+junitJupiter = "5.7.1"
+kotlin = "1.8.20"
+kotlinxCoroutines = "1.6.4"
+kotest = "4.2.3"
+ksp = "1.8.20-1.0.10"
+ktlint = "0.48.2"
+leakCanary = "2.10"
+material = "1.8.0"
+mockK = "1.13.4"
+mockk = "1.13.4"
```
Comment (thuutin) [inline gradle/libs.versions.toml:31]:
```suggestion
mockK = "1.13.4"
```

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+[libraries]
+androidx-compose-bom = { group = "androidx.compose", name = "compose-bom", version.ref = "androidxComposeBom" }
+androidx-compose-foundation = { group = "androidx.compose.foundation", name = "foundation" }
+androidx-compose-foundation-layout = { group = "androidx.compose.foundation", name = "foundation-layout" }
+androidx-compose-material-core = { group = "androidx.compose.material", name = "material"}
+androidx-compose-material-iconsExtended = { group = "androidx.compose.material", name = "material-icons-extended" }
+androidx-compose-runtime = { group = "androidx.compose.runtime", name = "runtime" }
+androidx-compose-ui-test = { group = "androidx.compose.ui", name = "ui-test-junit4" }
+androidx-compose-ui-testManifest = { group = "androidx.compose.ui", name = "ui-test-manifest" }
```
Comment (thuutin) [inline gradle/libs.versions.toml:62]:
```suggestion
androidx-compose-ui-test-manifest = { group = "androidx.compose.ui", name = "ui-test-manifest" }
```

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+kotlin-kapt = { id = "org.jetbrains.kotlin.kapt", version.ref = "kotlin" }
+nexus = { id = "io.github.gradle-nexus.publish-plugin", version.ref = "nexus" }
+spotless = { id = "com.diffplug.spotless", version.ref = "spotless" }
+
+[libraries]
+androidx-compose-bom = { group = "androidx.compose", name = "compose-bom", version.ref = "androidxComposeBom" }
+androidx-compose-foundation = { group = "androidx.compose.foundation", name = "foundation" }
+androidx-compose-foundation-layout = { group = "androidx.compose.foundation", name = "foundation-layout" }
+androidx-compose-material-core = { group = "androidx.compose.material", name = "material"}
+androidx-compose-material-iconsExtended = { group = "androidx.compose.material", name = "material-icons-extended" }
```
Comment (thuutin) [inline gradle/libs.versions.toml:59]:
```suggestion
androidx-compose-material-icons-extended = { group = "androidx.compose.material", name = "material-icons-extended" }
```

Comment (thuutin) [reply gradle/libs.versions.toml:59]: I guess we should be consistent with the lib names?

File: gradle/libs.versions.toml (hunk 3 above)
Comment (thuutin) [reply gradle/libs.versions.toml:62]: I guess we should be consistent with the lib names?

**Review by thuutin — APPROVED** [review]
LGTM. I just have 2 comments.

**Pull request merged**